backend/app/utils/jsonl_formatter.py

```
@@ -0,0 +1,354 @@
+            json_kwargs["separators"] = (',', ':')
+        
+        # 自定义分隔符
+        if self.config.custom_separators:
+            json_kwargs["separators"] = (
+                self.config.custom_separators.get("item_separator", ","),
+                self.config.custom_separators.get("key_separator", ":")
+            )
+        
+        return json.dumps(data, **json_kwargs)
+    
+    def _create_fallback_block(self, block: ParsedBlock, error_msg: str) -> Optional[str]:
```

Ruff (0.11.9)

Use `X | Y` for type annotations

Convert to `X | Y`

(UP007)